### ISSUE TITLE
[indent] Forbidden to use spaces when declaring a custom type with several types 

### ISSUE THREAD

**Issue by a-tarasyuk** [typescript-eslint/typescript-eslint#632]
**Repro**

```JSON
{
    "parser": "@typescript-eslint/parser",
    "parserOptions": {
        "sourceType": "module",
        "project": "./tsconfig.json"
    },
    "plugins": [
        "@typescript-eslint",
    ],
    "rules": {
        "indent": "off",
        "@typescript-eslint/indent": ["error", 4, {
            "SwitchCase": 1
        }],
    }
}
```

```TS
export type ProjectServiceEvent = LargeFileReferencedEvent
   | ProjectsUpdatedInBackgroundEvent
   | ProjectLoadingStartEvent
   | ProjectLoadingFinishEvent
   | ConfigFileDiagEvent
   | ProjectLanguageServiceStateEvent
   | ProjectInfoTelemetryEvent
   | OpenFileInfoTelemetryEvent;
```

**Expected Result**
  No indent error should keep the indentation of 4 spaces.

**Actual Result**
Expected indentation of **0** spaces but found 4.

**Versions**

| package                            | version |
| ---------------------------------- | ------- |
| `@typescript-eslint/eslint-plugin` | `1.10.2` |
| `@typescript-eslint/parser`        | `1.10.2` |
| `TypeScript`                       | `3.5.2` |
| `ESLint`                           | `5.16.0` |
| `node`                             | `8.15.0` |
| `npm`                              | `6.4.1` |

**Comment by j-f1** [comment]
You might be able to fix this by putting the first type reference onto the next line and adding a `|` in front of it, which is syntactically valid.

**Comment by j-f1** [comment]
I still think that this is a bug which should be fixed.

**Comment by bradzacher** [comment]
Duplicate of #121

This requires the indent rewrite to handle, because by default eslint completely ignores the indentation of binary operations.
